data-structures-queues/take-next-smallest.js

```diff
@@ -0,0 +1,26 @@
+/* exported takeNextSmallest */
+
+function takeNextSmallest(queue) {
+  if (typeof queue.peek() === 'undefined') return;
+
+  const container = [];
+  let lastVal = queue.dequeue();
+  container.push(lastVal);
+
+  let found = false;
+  while (typeof queue.peek() !== 'undefined' && !found) {
+    const last = container.length - 1;
+    if (container[last] < queue.peek()) {
+      lastVal = container[last];
+      found = true;
+    } else {
+      container.push(queue.dequeue());
```

Comment:
Why not just put it to the end of the queue here (`enqueue`)?

Reply:
I had it that way in my previous solution, but I couldn't figure out a way to exit the loop if no value was found that fit the problem's criteria. If we assume a strictly increasing sequence, wouldn't enqueue just keep adding values back into the queue?